test/unit/org/apache/cassandra/cql3/validation/operations/CQLVectorTest.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3.validation.operations;
+
+import org.junit.Test;
+
+import org.apache.cassandra.cql3.CQLTester;
+
+public class CQLVectorTest extends CQLTester.InMemory
+{
+    @Test
+    public void select()
+    {
+        createTable(KEYSPACE, "CREATE TABLE %s (pk vector<int, 2> primary key)");
+
+        execute("INSERT INTO %s (pk) VALUES ([1, 2])");
+
+        assertRows(execute("SELECT * FROM %s WHERE pk = [1, 2]"), row(list(1, 2)));
```

Comment:
`list(1, 2)` and `row(list(1, 2)` are repeated many times across the test. We can put them in a var:
```java
Vector<Integer> vector = vector(1, 2);
Object[] row = row(vector);

assertRows(execute("SELECT * FROM %s WHERE pk = [1, 2]"), row);
assertRows(execute("SELECT * FROM %s WHERE pk = ?", vector), row);
assertRows(execute("SELECT * FROM %s WHERE pk = [1, 1 + 1]"), row);
assertRows(execute("SELECT * FROM %s WHERE pk = [1, ?]", 2), row);
...

```